core/lib.rs

```diff
@@ -893,6 +893,14 @@ impl Connection {
         self.page_size.get()
     }
 
+    pub fn get_database_path(&self) -> Option<String> {
+        Some(self._db.path.clone())
```

Comment:
This doesn't need to be an `Option` if we always return `Some`. Also, can we return a reference?

Reply:
I'll just move the caller logic here.
I wanted to return a string because this can be "". But I ended up moving the logic to test that outside of the function. But I am having second thoughts on that.